raiden/network/transport/matrix/client.py

```diff
@@ -361,6 +365,7 @@ def stop(self) -> None:
         self.sync_token = None
         self.should_listen = False
         self.rooms: Dict[str, Room] = {}
+        self._worker_pool.join(raise_error=True)
```

Comment:
I guess we could just kill the pool instead, but that would require discarding the token and forcing a new login, to get the fresh presence data.